homeassistant/components/nuki/sensor.py

```diff
@@ -0,0 +1,67 @@
+"""Battery sensor for the Nuki Lock."""
+
+from homeassistant.components.sensor import SensorDeviceClass, SensorEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import NukiEntity
+from .const import ATTR_NUKI_ID, DATA_COORDINATOR, DATA_LOCKS, DOMAIN as NUKI_DOMAIN
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up the Nuki lock sensor."""
+    data = hass.data[NUKI_DOMAIN][entry.entry_id]
+    coordinator = data[DATA_COORDINATOR]
+
+    async_add_entities(
+        [NukiBatterySensor(coordinator, lock) for lock in data[DATA_LOCKS]]
+    )
+
+
+class NukiBatterySensor(NukiEntity, SensorEntity):
+    """Representation of a Nuki Lock Battery sensor."""
+
+    _attr_has_entity_name = True
+
+    @property
+    def name(self):
+        """Return the name of the lock."""
+        return "Battery"
```

Comment:
Please use the `_attr_` attributes to set the constant properties. Eg instead of defining a `name` property we set the class attribute `_attr_name`.

Reply:
Thanks for your feedback. I adapted the code.

Additional question: can I do the same for the ones that return data from the `pynuki` package?

Can this:
```python
@property
def extra_state_attributes(self):
    """Return the device specific state attributes."""
    data = {
        ATTR_NUKI_ID: self._nuki_device.nuki_id,
    }
    return data
```

Be turned into this?

```python
_attr_extra_state_attributes = {
  ATTR_NUKI_ID: self._nuki_device.nuki_id,
}
```

(Sorry, I'm new to HA development and haven't much experience with Python)

Reply:
Yes, but then we need to add an `__init__` method as it's bad practice to set a mutable object as a class attribute.

Also, this will only work if the attribute won't need updating after the entity is created. If the attribute needs updating, we either need to do that in an state update callback or keep using a property that is read when the entity state is updated.